### PR TITLE
Stand Arrow 2: Electric Bogaloo

### PR DESCRIPTION
**Pull request by yobob591**
This time with files!

Creates a stand arrow in space which behaves as a holoparasite injector with infinite uses, but with a catch; there is a 60% chance of killing the user. It can also be used on others, but remember you are liable for their death.

I also forgot to mention you cant choose your holopara, you get what you get

### CODE REVIEW

**Comment by yobob591** [comment]
what did i fuck up

**Comment by yobob591** [comment]
im gonna need to be babysat ive never used git before

**Comment by trifornt-tinfornt** [comment]
60% chance of killing people. Or at least 50%. Holoparasites are 20 TC traitor items, and seeing the owl suit and how people rushed every round to grab that makes me think that even with a chance to instantly kill them that's a gamble they'd gladly take.

**Comment by yobob591** [comment]
I meant you have a 40% chance of getting a stand, im dumb

**Comment by yobob591** [comment]
I think i fixed the shit

**Comment by yobob591** [comment]
fucccccc

**Comment by Pyko1** [comment]
No need to commit  tgstation.dme and _maps/templates/z4/backup/DJ_Station_18x23.dmm

**Comment by yobob591** [comment]
ah

**Comment by yobob591** [comment]
fuck me how do i make it not commit tgstation.dme now
i cant edit it correctly
trying again

**Comment by yobob591** [comment]
fyckmfisngaijanfaws

kill me

**Comment by yobob591** [comment]
can i just delete them

**Comment by yobob591** [comment]
literally wont let me revert anymore what the FUCK

**Comment by yobob591** [comment]
ive ruined it aaaaaaaaaaaaaaaaaaaaaaaaaa

**Comment by Pyko1** [comment]
you can delete the backup map, and then overwrite the tgstation.dme with the one we have on server.

**Comment by yobob591** [comment]
how

**Comment by yobob591** [comment]
how do i pick only file

**Comment by yobob591** [comment]
i fucking broke it so hard

**Comment by Pyko1** [comment]
What client you use?

**Comment by yobob591** [comment]
github desktop

**Comment by Pyko1** [comment]
When you commit shit you can choose what files you commit.

And you can go to the files and manually edit/remove files.

**Comment by yobob591** [comment]
yes i picked some bad files now it wont let me not pick them when i revert and revert reverts

**Comment by Pyko1** [comment]
you have to commit over them, Ie remove the backup file manually and overwrite the tgstation with the one we have on server, and then commit.

**Comment by yobob591** [comment]
commit over them with what?

**Comment by yobob591** [comment]
also how overwrite dme

**Comment by yobob591** [comment]
waaait

**Comment by Pyko1** [comment]
Well now you overwrote everything with hippie master so you just need to toss the right files back.
And then ensure you only commit the right files.

**Comment by yobob591** [comment]
that is gonna be the hard part since it wont let me select individual files

**Comment by Pyko1** [comment]
Just download the files, and copy paste them at correct places in file explorer.

And check the modified files before you hit the commit button.

**Comment by yobob591** [comment]
finally

**Comment by yobob591** [comment]
praise the lord

**Comment by yobob591** [comment]
the arrow doesnt spawn in dj station reeeeeeeeeeeeeeeeeeeeee

**Comment by yobob591** [comment]
gonna have to ask for help with that tomorrow

**Comment by Pyko1** [comment]
You edited 2 of em at the same spot on the random generation template.
"aj" = (/obj/item/weapon/guardiancreator/standarrow,/obj/item/weapon/guardiancreator/standarrow,/turf/simulated/floor/plating,/area/djstation)

But I believe our DJ station still static part of Z4.

**Comment by Pyko1** [comment]
That template/z4 folder is currently un-used and containing the Z4 structures as invidual templates.

**Comment by yobob591** [comment]
Oh thank

**Comment by yobob591** [comment]
I changed the location to somewhere more fitting

**Comment by yobob591** [comment]
lol dids it again

**Comment by Pyko1** [comment]
Might want to remove that double arrow from template even tho its not used currently.

**Comment by Pyko1** [comment]
Also, 

1. Rush to get arrow
2. Build Auto-cloner
3. Use arrow
4. ???
5. Profit

**Comment by Pyko1** [comment]
Also I can already see a guy getting back to station with holo, and everyone trying to kill him for having it, and him killing everyone in "selfdefence". As well as possibly poking them with arrow, as even if they die to it it was selfdefence and if not, now they know you just had found the arrow.

**Comment by GuyonBroadway** [comment]
That'd probably fall under banbating as shit.

**Comment by Pyko1** [comment]
Hardly ban baiting really, especially when there is 3 sides involved.

Anyone who sees your holo can kill you for having "antag item".
The holo is ultimately tasked to protect your life.
And you are also rather interested in stopping the guy from killing you or your holo.

Also I recall there is a small delay on getting your guardian after you inject yourself, due to polling ghosts and shit, but the kill effect is instant.
So basically you could use it as 60% instant kill item, and even if it does not work you have time to take care of the guy before he gets his holo.
And that way he also gets gibbed.

**Comment by GuyonBroadway** [comment]
By which I mean a non-antag using it to kill half the station in "self defence". That would be banable as fuck

**Comment by Pyko1** [comment]
Well things tend to escalate, and at which point you have to let them kill you without fighting back?
Because people tend to try to lynch others with antag stuff, and if anyone sees your holo, or you actually killing someone with it (even if in selfdefence) you are in a world of LYNCH!

**Comment by GuyonBroadway** [comment]
I think it's a case of "if you use a holopara as a non-antag you acept the risks and almost catban like status of other people being able to kill you."

**Comment by Pyko1** [comment]
Yes but the holo does the killing usually not the master. 
So ban the holoparasite for defending his master, or ban the holoparasite for refusing to defend his master?

**Comment by Pyko1** [comment]
Not to mention holos will most likely assume their master is antag unless actually told otherwise.
Since they get no alert what so ever that they weren't actually spawned with holoparasite injector.

**Comment by JohnGinnane** [comment]
Is this ready to be reviewed?

**Comment by yobob591** [comment]
Yes its all working, also pyko holoparasites should no longer make you valid since they wont be a traitor only item anymore

**Comment by yobob591** [comment]
We used to have space para injectors and it worked great, they were only removed along side a ton of other stuff in the telescience nerf

**Comment by Spacedong** [comment]
or maybe you could make the stands gained from the stand arrow use this old sprite: 
https://vgy.me/rejHKL.gif

just a thought

**Comment by yobob591** [comment]
oh good idea, ill do that now

**Comment by Spacedong** [comment]
cool

**Comment by blobbernaut** [comment]
You can just put things in wall mounted safes or in vaults to keep people from telesciencing it.

**Comment by yobob591** [comment]
Buffed kill chance to 70% by popular demand to prevent powergaming
Stands have different sprites to stop balid salib
Belt sprite

**Comment by blobbernaut** [comment]
it won't prevent powergaming, it'll just be used as a weapon now

an instant kill weapon

**Comment by trifornt-tinfornt** [comment]
70% is too high. 50% or 60%.

**Comment by yobob591** [comment]
Lets meet in the middle with 60

**Comment by yobob591** [comment]
it was 40

**Comment by yobob591** [comment]
I may add syndies around it to make it harder to get if enough people want that

**Comment by ArcLumin** [comment]
update your changelog

**Comment by blobbernaut** [comment]
50% is fine, it just needs: 

Guards - Russians with mosins work best, high damage bullets and the gun itself is shitty. 
Inside something - A wall safe works best, can't get it with telesci. 
Not quick to get - It should not be in a straight line with anything else in space.

**Comment by yobob591** [comment]
Weve decided to make it a mining spawn item and remove the husk chance

**Comment by Kierany9** [comment]
It just needs to not spawn consistently where it can be rushed every round. If it spawns on one of the random templates, or as an item spawned in mining, then there won't be any issues with powergaming.

**Comment by ArcLumin** [comment]
Okay a few things;

1. As far as I can tell, this does not cause someone to get a stand if it's thrown at them. Do you intend to make that a feature, or too OP?

2. This update allows any simple mobs to get stands. This causes some hilariousness when Ian gets a stand, but what's stopping me from using it on my current stand so my stand gets a stand?

**Comment by yobob591** [comment]
I considered that but i decided due to a kill chance that would be kinda powerful, maybe in the future

i dont know ive never tried that before and it sounds amazing, i have no idea how to stop that from happening either ive never coded for byond before so someone else is gonna have to fix that or at least guide me

**Comment by ArcLumin** [comment]
let stands get stands because you still have the killchance.

**Comment by yobob591** [comment]
lets do it, that might end up being hilarious

**Comment by ArcLumin** [comment]
Might I mention that where ever it is, it should appear every round, but in a random area

**Comment by ArcLumin** [comment]
As a side note we should fine a way to stop the github slackbot from 100% using the text used on github for @everyone

**Comment by yobob591** [comment]
I set it to the wizden mining template, I could add it to another template to increase its spawn chance but right now its not every round

**Comment by ArcLumin** [comment]
boost the chance for stuff to spawn inside of it and that's fine.









@channel testing

**Comment by yobob591** [comment]
I dont exactly know how, ive never done asteroid template shit

**Comment by ArcLumin** [comment]
I can walk you through it later, kinda busy now, or I can do it in a seperate change

**Comment by yobob591** [comment]
seperate change sounds good, im still adjusting damage types and shit right now so it doesnt husk

**Comment by QuarianCommando** [comment]
This looks like fun, and needs to happen. I miss having the old guardian spirits in space.

**Comment by yobob591** [comment]
Right is partly added to mining and no longer husks

**Comment by yobob591** [comment]
What should the rules for the stand be for when a simple mob that isnt player controlled i.e. Ian gets a stand?

**Comment by ArcLumin** [comment]
Same as anything else. Simple mobs are always valid salid, as per usual.

**Comment by ArcLumin** [comment]
Also DRONES WITH STANDS BOYS

**Comment by Kierany9** [comment]
>### Also DRONES WITH STANDS BOYS

## STAND HATS!!!

# YOU CAN WEAR FUCKING STAND USERS ON YOUR HEADS!!!

**Comment by yobob591** [comment]
currently blacklisting revenant and statue cause invincible stands wew lad

**Comment by yobob591** [comment]
and horror form ling o fuck

**Comment by ArcLumin** [comment]
>wondering where this stand keeps coming from

>enemy user is on your head

**Comment by ArcLumin** [comment]
Needs the blacklist, then the code will be good

**Comment by McBawbaggings** [comment]
Make it human only please.
Animal stand users is too memey.

Also, make attacking someone with it a channel thing so you can't insta kill cunts with it.

**Comment by ArcLumin** [comment]
Animal stand users is actually in jojo, multiple times. And you can already be a monkey or animal stand user in game, with some effort.

Also yeah, having a certain amount of time required to use the arrow would be fine

**Comment by yobob591** [comment]
mcbaw I tried to make a cooldown but its bein shit, ill take a look at channels like ei nath and go from there

**Comment by yobob591** [comment]
Also blacklist is still in development

**Comment by yobob591** [comment]
Right, now theres an approx 3 second channel for stabbing, still need FUCKING BLACKLIST

**Comment by yobob591** [comment]
Working blacklist after 9 years in development

**Comment by yobob591** [comment]
Right all thats left is getting this thing to spawn consistently in space

**Comment by yobob591** [comment]
none of these commits have gone anywhere

i just get https://github.com/yobob591/HippieStation13/commit/0d97a12fcce2fd1f7ffa18897219f3d95e0fc033

**Comment by yobob591** [comment]
help

**Comment by yobob591** [comment]
nvm fixed

**Comment by yobob591** [comment]
Right, added new roomtype to asteroid and buffed spawn rate to ensure it gets placed correctly each round. Just need input from @McBawbaggings and @Hackole

**Comment by yobob591** [comment]
also thanks to @ArcLumin for helping me get the asteroid shit down

**Comment by yobob591** [comment]
I did some testing and this would probably take anywhere from 5 to 30 minutes of mining to find depending of if you get lucky with the spawn or not

**Comment by oprayx77** [comment]
hey the suit got added so you could shoehorn that into the room aswell

**Comment by yobob591** [comment]
right

**Comment by yobob591** [comment]
Added kiras suit to le room

**Comment by yobob591** [comment]
I got bored of waiting so I made it play the opening riff of Stand Activated to the person who gets a stand when they get it

**Comment by Swagile** [comment]
giving my GM +1 for a two week test run due to concerns on admin slack of abuse

**Comment by Dudeinyourbackyard** [comment]
KIMI NO STANDO POWA

yeah ok 1/3

**Comment by yobob591** [comment]
2/3

**Comment by yobob591** [comment]
wew

**Comment by yobob591** [comment]
https://www.youtube.com/watch?v=WIkhGvXB0ek

**Comment by ArcLumin** [comment]
I've been told to merge SO LETS MERGE

**Comment by KittyKathy** [comment]
Just some thing to think you should know.
One thing you should know that since it does toxin damage now when you fail the chance to get it you can defibrillate the person to revive them easily and add give them some antitoxin for like infinite retries. 
You can also probably self revive yourself with a health sensor grenade that has strange reagent and healing chemicals. 
Vampires can not die from any damage that is not burns if they have clean blood it will stop them from dying and they can use their regeneration power or any chemical to get back from critical condition and try again. Even if they do manage to die somehow vampire can come back to life by turning into a bat after healing with their regeneration abilities which work when dead.
Also the super important thing you should know is it is impossible to kill a vampire who has a stand because they literally can not die if they have clean blood unless you set them on fire or something but they can turn into a bat and fly away from any stun anytime and they can't take damage in bat mode if you attack their stand because it gives their body the god mode flag while in bat mode form
I was too late it was merged like 2 minutes ago

**Comment by yobob591** [comment]
never thought about vamps, probably should fix this in the future, the revives are fine since its hard enough to find the arrow anyways

**Comment by Swagile** [comment]
the problem is that vampire is bugged anyway

and the chances of it being a vampire round + finding this thing + using it are so rare that you might as well be rewarded for it

a fix can be made, but i don't see it being a problem

**Comment by ArcLumin** [comment]
When did it get changed from burn to toxin?

Also, if you die with a holopara you're instagibbed. So, good luck healing that when you're dead.

**Comment by yobob591** [comment]
I forgot about that, also i change it to toxin to stop the husking a ways back, we decided not to husk anymore cause its in a random spot on mining

**Comment by yobob591** [comment]
also vamps with stands should be strong due to the rarity of it happening and because that happened in jojo

**Comment by ArcLumin** [comment]
Plus, it's not like you wouldn't be able to not do it in a normal round, without a stand arrow

**Comment by KittyKathy** [comment]
Oh yeah that is actually good that when a vamp dies with a stand they get gibed so they can't revive from that unless they have a auto cloner set up or got scanned before they died.

**Comment by yobob591** [comment]
and vamps that are cloned lose their powers and their stand

**Comment by KittyKathy** [comment]
That is probably a bug that they lose their vampire powers, I think if you monkey them they also lose their vampire powers but yeah I have actually went up against a vampire that was a traitor and had a holoparasite I was the detective modified my revolver to shoot .357s and I shot them enough to kill 4 people but they ended up killing me because they could not die and there was nothing i could do to stop the stand.
I never watched this Jojo thing but yeah i guess it's okay since it should be rare lets see what happens.

**Comment by ArcLumin** [comment]
Not a bug. Vampire's powers are tied to their body, not their mind.

**Comment by yobob591** [comment]
yeah in jojo a vampire gets a stand and is nearly invincible

**Comment by yobob591** [comment]
Also i tested if an arrow rejects a vamp it insta max crits them and unless they have blood they are trapped in eternal crit

**Comment by yobob591** [comment]
scratch that, near eternal crit, they CAN die from toxins it seems but it takes ages

**Comment by McBawbaggings** [comment]
I only see 2 GM inputs and Spank HA input. This got merged like that?

**Comment by Spacedong** [comment]
@McBawbaggings this never should have been merged in the first place, forcemerge the removal of it.

**Comment by blobbernaut** [comment]
this is good though please keep

**Comment by yobob591** [comment]
its going to spawn from meteors now

**Comment by yobob591** [comment]
so dont worry its not going away

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 1]
```diff
@@ -728,9 +817,37 @@
...
 	failure_message = "<B>...but soon settles again. Guess they weren't ready to hatch after all.</B>"
+	limiteduses = TRUE
 
 /obj/item/weapon/guardiancreator/biological/choose
 	random = FALSE
+	limiteduses = TRUE
+
+/obj/item/weapon/guardiancreator/standarrow
+	name = "Stand Arrow"
+	desc = "A mysterious arrow capable of granting great power. Be carfeul, there is a chance it won't take to you..."
```
Comment (FuryMcFlurry) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm:828]: "A mysterious arrow capable of granting great power. Be carfeul, there is a chance it won't take to you..."
spellcheck please.

**Comment by yobob591** [comment]
i fixed this fucking ages ago

**Comment by FuryMcFlurry** [comment]
mybad dude.
